### DISCUSSION TITLE
Reject messages from MessageHandler

### DISCUSSION THREAD

**Discussion by bard83** [Farfetch/kafkaflow#530] [Q&A]
Hi,

based on this [discussion](https://github.com/Farfetch/kafkaflow/discussions/347) I was wondering how to handle the opposite situation. To be more precise, in my MessageHandler, I would like to deadletter non valid messages. I assumed that the handler is clever enough to do it whenever an exception is raised. Is my assumption right? In case not, is there a way to deadletter the message?
Below a snippet to have a better idea:

```csharp
public class ExampleHandler : IMessageHandler<MyExampleMessage>
{
    public Task Handle(IMessageContext context, MyExampleMessage message)
    {
           if (!IsValidMessage(message))
           {
                   throw new InvalidOperationException("Message received is not valid");
           }

          // do other stuff.
          return Task.CompletedTask;
    }
}
```

thanks in advance

**Comment by JoaoRodriguesGithub** [comment]
The dead letter that you refer to is when an error occurs, the message goes for a dead letter queue topic?

For that, I can confirm that KafkaFlow does not have that behavior by default, but that does not mean that it can't be implemented on the middleware by catching the exception and producing the message to another topic.

**Reply by bard83** [reply]
> the message goes for a dead letter queue topic?

Yes, I do.

Then comes the question. What does happen to the message when the `Handle` throws exception? Will be reworked from the same topic?
